test/test_mapper_llvm.cc

```diff
@@ -66,6 +65,176 @@ def fun(float(N, M) A, float(N, M) B) -> (C) {
+  %39 = bitcast %struct.__cilkrts_stack_frame* %__cilkrts_sf to i8*
+  %sunkaddr23 = getelementptr i8, i8* %39, i64 48
+  %40 = bitcast i8* %sunkaddr23 to i8**
+  store volatile i8* %38, i8** %40, align 8
+  %41 = call i32 @llvm.eh.sjlj.setjmp(i8* nonnull %28) #3
+  %42 = icmp eq i32 %41, 0
+  br i1 %42, label %cilk.sync.runtimecall.i, label %cilk.sync.excepting.i
+
+cilk.sync.runtimecall.i:                          ; preds = %cilk.sync.savestate.i
+  call void @__cilkrts_sync(%struct.__cilkrts_stack_frame* nonnull %__cilkrts_sf) #0
+  br label %__cilk_sync.exit
+
+cilk.sync.excepting.i:                            ; preds = %cilk.sync.savestate.i
+  %43 = bitcast %struct.__cilkrts_stack_frame* %__cilkrts_sf to i32*
+  %44 = load volatile i32, i32* %43, align 8
+  %45 = and i32 %44, 16
+  %46 = icmp eq i32 %45, 0
+  br i1 %46, label %__cilk_sync.exit, label %cilk.sync.rethrow.i
+
+cilk.sync.rethrow.i:                              ; preds = %cilk.sync.excepting.i
+  call void @__cilkrts_rethrow(%struct.__cilkrts_stack_frame* nonnull %__cilkrts_sf) #4
+  unreachable
+
+__cilk_sync.exit:                                 ; preds = %loop_exit, %cilk.sync.runtimecall.i, %cilk.sync.excepting.i
+  %47 = bitcast %struct.__cilkrts_stack_frame* %__cilkrts_sf to i32*
+  %48 = bitcast %struct.__cilkrts_stack_frame* %__cilkrts_sf to i8*
+  %sunkaddr24 = getelementptr i8, i8* %48, i64 16
+  %49 = bitcast i8* %sunkaddr24 to %struct.__cilkrts_worker**
+  %50 = load volatile %struct.__cilkrts_worker*, %struct.__cilkrts_worker** %49, align 8
+  %51 = getelementptr inbounds %struct.__cilkrts_worker, %struct.__cilkrts_worker* %50, i64 0, i32 12, i32 0
+  %52 = load i64, i64* %51, align 8
+  %53 = add i64 %52, 1
+  store i64 %53, i64* %51, align 8
+  %54 = load volatile %struct.__cilkrts_worker*, %struct.__cilkrts_worker** %49, align 8
+  %55 = bitcast %struct.__cilkrts_stack_frame* %__cilkrts_sf to i8*
+  %sunkaddr25 = getelementptr i8, i8* %55, i64 8
+  %56 = bitcast i8* %sunkaddr25 to %struct.__cilkrts_stack_frame**
+  %57 = load volatile %struct.__cilkrts_stack_frame*, %struct.__cilkrts_stack_frame** %56, align 8
+  %58 = getelementptr inbounds %struct.__cilkrts_worker, %struct.__cilkrts_worker* %54, i64 0, i32 9
+  store volatile %struct.__cilkrts_stack_frame* %57, %struct.__cilkrts_stack_frame** %58, align 8
+  store volatile %struct.__cilkrts_stack_frame* null, %struct.__cilkrts_stack_frame** %56, align 8
+  %59 = load volatile i32, i32* %47, align 8
+  %60 = icmp eq i32 %59, 16777216
+  br i1 %60, label %__cilk_parent_epilogue.exit, label %body.i
+
+body.i:                                           ; preds = %__cilk_sync.exit
+  call void @__cilkrts_leave_frame(%struct.__cilkrts_stack_frame* nonnull %__cilkrts_sf) #0
+  br label %__cilk_parent_epilogue.exit
+
+__cilk_parent_epilogue.exit:                      ; preds = %__cilk_sync.exit, %body.i
+  ret void
+}
+)LLVM";
+  EXPECT_EQ(correct_llvm, toString(mod->getFunction("kernel_anon")));
```

Comment:
This seems overly ambitious :), care to relax a bit?
Also, we only use vanilla LLVM internally atm so this is not a portable test, can you please separate test_mapper_llvm from test_mapper_tapir which should only compile conditionally?

Reply:
Resolve for both. However, as after this our CPU backend starts emitting parallel code [and we start doing some scheduling for CPU], it may be wise to nudge the internal folks to add Tapir :P